### DISCUSSION TITLE
Azure SSO

### DISCUSSION THREAD

**Discussion by wearethedynamite** [9p4/jellyfin-plugin-sso#121] [Q&A] [answered]
I had everything working and was able to authenticate using my microsoft credentials, created a nice login page and was happy for a time. Then went to sign in one day and the redirect uri seems to drop https to http, meaning it doesn't match that set in the azure application.
Not sure when this break happened, as my sessions generally stay logged in, I was just logging in on a new device when I noticed.

Wondering if anyone has any insight as to why it's presenting a http redirect? I even force ssl in NGINX...
I am using nginx proxy manager and cloudflare tunnels if that makes a difference?

Thanks for your time.

**Comment by 9p4** [comment]
Does your reverse proxy forward `X-Forwarded-Proto`?

**Reply by wearethedynamite** [reply]
I hadn't specifically added it, but adding it in to test doesn't seem to resolve the issues.
Though according to this - https://github.com/NginxProxyManager/nginx-proxy-manager/issues/1556 - that header is set automatically anyway.
I can't for the life of me figure out what has changed from when It was working until now!

**Reply by 9p4** [reply]
What plugin version are you using? Is there anything in the logs?

**Reply by wearethedynamite** [reply]
Version reports as 3.4.0.0 on jellyfin, theres a lot of "sloww http response for http://my-server.org errors and  
 Emby.Server.Implementations.HttpServer.WebSocketManager: WS "172.18.0.1" request lines.
But i've been messing around with it a lot today so it's a big log file and I don't really know what to look  for.
[log_20230609.log](https://github.com/9p4/jellyfin-plugin-sso/files/11708713/log_20230609.log)

**Reply by 9p4** [reply]
Could you clear the log, restart Jellyfin, make one request, then save the log and send it to me?

**Reply by wearethedynamite** [reply]
[log_20230609.log](https://github.com/9p4/jellyfin-plugin-sso/files/11708751/log_20230609.log)
Fresh log file, thanks for your help!

**Reply by 9p4** [reply]
Are you sure you made one login request? I don't see anything in the logs.Are you sure you made one login request? I don't see anything in the logs.

**Reply by wearethedynamite** [reply]
Yep, the final line Jellyfin.Plugin.SSO_Auth.Api.SSOController: SSO Controller initialized appears when I click my login with microsoft button - which links to https://my-jellyfin-server.org/sso/oid/p/microsoft.  
I put in my credentials/approve authenticator request - login request is successfull in azure logs then get this.
![image](https://github.com/9p4/jellyfin-plugin-sso/assets/79176174/e2ba6db7-af3a-4662-b31c-e7b4fa0e3ca7)

**Reply by 9p4** [reply]
Would you be willing to compile from the latest commit and see if that fixes your problem (https://github.com/9p4/jellyfin-plugin-sso/pull/71)?

**Reply by 9p4** [reply]
Also see if https://github.com/9p4/jellyfin-plugin-sso/issues/79#issuecomment-1253280570 might fix your issue

**Reply by wearethedynamite** [reply]
> Also see if [#79 (comment)](https://github.com/9p4/jellyfin-plugin-sso/issues/79#issuecomment-1253280570) might fix your issue
Tried this with no luck unfortunately.

Happy to compile from the latest commit, jut not too sure how to do so. 😅

**Reply by wearethedynamite** [reply]
> > Also see if [#79 (comment)](https://github.com/9p4/jellyfin-plugin-sso/issues/79#issuecomment-1253280570) might fix your issue
> > Tried this with no luck unfortunately.
> 
> Happy to compile from the latest commit, jut not too sure how to do so. 😅

Just tested the nightly build and same result...

**Reply by 9p4** [reply]
> > Also see if [#79 (comment)](https://github.com/9p4/jellyfin-plugin-sso/issues/79#issuecomment-1253280570) might fix your issue
> Tried this with no luck unfortunately.

Could I take a look at your nginx config?

**Reply by wearethedynamite** [reply]
I've actually just migrated to Traefik, in part because I'd hoped it would resolve this, but i'd been meaning to so so for a while anyway.
Unfortunately I still have the same issue, despite forcing https through Traefik.  I'm wondering if there's something obvious in the jellyfin config that I've missed.... Is there any way to force a redirect URI? what's the process in which it get's generated?

I've also tried disabling the CF tunnel and adding DNS A records/port forwarding.
I had this working lovely when I set it up initially but now i've no idea what i have messed up. 😅

**Reply by 9p4** [reply]
Could I take a look at your traefik config?

**Reply by wearethedynamite** [reply]
Sure thing, I mostly followed this guide from Ibracorp.
https://docs.ibracorp.io/traefik/master/unraid

traefik.yml
https://pastebin.com/nuevYL0b

fileconfig.yml
https://pastebin.com/rri6aiTG

in my jellyfin docker container I have these set lables;
![image](https://github.com/9p4/jellyfin-plugin-sso/assets/79176174/2d3e1d25-3998-4500-b600-66806e8593dd)

**Reply by 9p4** [reply]
Can you follow the guide [here](https://jellyfin.org/docs/general/networking/traefik2) and set all of those traefik values too?

**Reply by wearethedynamite** [reply, marked as answer]
I have finally solved this. Turns out the Jellyfin setting for known proxies wasn't working when using hostnames - despite being on the same custom docker network.
Adding the IP for Traefik would have probably been ok, but as that would be dynamic - adding the ip range of the custom network seems to have fixed it. 
Sorry for being so dumb and thank you so much for your time in trying to resolve this.

**Reply by 9p4** [reply]
Glad you figure it out! If you need any more help, feel free to ask!
